envoy/network/BUILD

```diff
@@ -63,6 +63,22 @@ envoy_cc_library(
     deps = ["//envoy/network:address_interface"],
 )
 
+envoy_cc_library(
+    name = "dns_factory_interface",
+    hdrs = ["dns_factory.h"],
+    deps = [
+        ":dns_interface",
+        "//envoy/api:api_interface",
+        "//envoy/event:dispatcher_interface",
+        "@envoy_api//envoy/config/bootstrap/v3:pkg_cc_proto",
+        "@envoy_api//envoy/config/cluster/v3:pkg_cc_proto",
+        "@envoy_api//envoy/extensions/common/dynamic_forward_proxy/v3:pkg_cc_proto",
+        "@envoy_api//envoy/extensions/filters/udp/dns_filter/v3alpha:pkg_cc_proto",
+        "@envoy_api//envoy/extensions/network/dns_resolver/apple/v3:pkg_cc_proto",
```

Comment:
Why is this linking specific extension protos to the interface lib?

Reply:
moved it to source/comment/network/dns_resolver